classes/metric/base.php

```diff
@@ -154,4 +177,37 @@ public function set_last_generate_time(int $timestamp) {
      * @return metric_item
      */
     abstract public function generate_metric_item(int $starttime, int $finishtime): metric_item;
+
+    /**
+     * Metric's ability to be backfilled.
+     *
+     * @return bool
+     */
+    public function is_backfillable(): bool {
+        return false;
+    }
+
+    /**
+     * Returns records for backfilled metric.
+     *
+     * @param int $backwardperiod Time from which sample is to be retrieved.
+     * @param int $finishtime If data is being completed argument is passed here.
+     *
+     * @return array|null
+     */
+    public function generate_metric_items(int $backwardperiod, int $finishtime = null): ?array {
```

Comment:
Not quite sure what you mean by plain days. To me if the metric happens to be once a day then it would be time slices of each day and each day starts at midnight and ends at midnight. But this all depends on the metric frequency, the same applies if it is an hourly metric or every 5 minutes